### PR TITLE
Fixes issues where extraction queries can't render tables

### PR DESCRIPTION
**Pull request by joannecheng**
## What does this PR do? How does it affect users?

Currently users cannot view extraction previews in explorer. This is because `getChartTypeOptions` was returning the wrong `dataType`. To fix this issue, I moved the `else if` statements that looked at the `analysis_type` to the top of the block. I also moved the functions that determined chart types to a separate file to make for easier maintainability and unit testing (we probably need a lot more unit testing for these functions).

## How should this be tested?

- [x] First, check out master. Run an extraction and confirm that table visualizations are broken for extraction queries.
- [x] Check out this branch. Run an extraction and confirm that table visualizations are present.

Step through the code line by line. Things to keep in mind as you review:
 - Are there any edge cases not covered by this code?
 - Does this code follow conventions (naming, formatting, modularization, etc) where applicable?

Fetch the branch and/or deploy to staging to test the following:

- [x] Does the code compile without warnings (check shell, console)?
- [x] Do all tests pass?
- [x] Does the UI, pixel by pixel, look exactly as expected (check various screen sizes, including mobile)?
- [x] If the feature makes requests from the browser, inspect them in the Web Inspector. Do they look as expected (parameters, headers, etc)?
- [x] If the feature sends data to Keen, is the data visible in the project if you run an extraction (include link to collection/query)?
- [x] If the feature saves data to a database, can you confirm the data is indeed created in the database?

### CODE REVIEW

File: client/js/app/utils/ChartTypeUtils.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+var _ = require('lodash');
+
+module.exports = {
+  getQueryDataType: function(query){
+    var isInterval = typeof query.interval === "string";
+    var isGroupBy = typeof query.group_by === "string" || (query.group_by instanceof Array && query.group_by.length === 1);
+    var is2xGroupBy = query.group_by instanceof Array;
+    var dataType;
```
Comment (joannecheng) [inline client/js/app/utils/ChartTypeUtils.js]: Moved each `var` declaration to own line, as specified in airbnb style guide: https://github.com/airbnb/javascript/tree/master/es5#variables

Comment (aroc) [inline client/js/app/utils/ChartTypeUtils.js]: ++ on that. I have a strong preference for this!

**Comment by aroc** [comment]
Fantastic. Let's go on a cruise. :ship:
